pkg/vmimage/vmimage_test.go

```diff
@@ -0,0 +1,37 @@
+package vmimage
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/openshift/openshift-azure/test/util/tls"
+)
+
+func TestGenerateTemplate(t *testing.T) {
```

Comment:
good, although this actually boils down all the way to:
```golang
func TestGenerateTemplate(t *testing.T) {
	builder := Builder{
		SSHKey:     tls.GetDummyPrivateKey(),
		ClientKey:  tls.GetDummyPrivateKey(),
		ClientCert: tls.GetDummyCertificate(),
	}

	_, err := builder.generateTemplate()
	if err != nil {
		t.Error(err)
	}
}
```

Reply:
if you fix the top of populate.DummyCertsAndKeys as follows:
```golang
func DummyCertsAndKeys(v interface{}) {
	var walk func(v reflect.Value)

	walk = func(v reflect.Value) {
		if v.Type().PkgPath() != "" &&
			(!strings.HasPrefix(v.Type().PkgPath(), "github.com/openshift/openshift-azure/") ||
				strings.HasPrefix(v.Type().PkgPath(), "github.com/openshift/openshift-azure/vendor/")) {
			return
		}
```

then this reduces to:
```golang
func TestGenerateTemplate(t *testing.T) {
	var builder Builder
	populate.DummyCertsAndKeys(&builder)

	_, err := builder.generateTemplate()
	if err != nil {
		t.Error(err)
	}
}
```

Reply:
(@thekad if this was black magic - worth reading through the function, understanding it and adding a docstring.  We already use it a lot and are going to use it more).